test/test_t1.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Test T1 experiment
+"""
+
+import unittest
+
+import numpy as np
+
+from qiskit.providers.aer import QasmSimulator
+from qiskit.providers.aer.noise.errors.standard_errors import thermal_relaxation_error
+from qiskit.providers.aer.noise import NoiseModel
+
+from qiskit_experiments.composite import BatchExperiment, ParallelExperiment
+from qiskit_experiments.characterization import T1Experiment
+
+backend = QasmSimulator()
+instruction_durations = \
+                      [('measure', [0], 3, 'dt'),
+                       ('x', [0], 3, 'dt')]
+gate_time = 0.1
+
+# Fix seed for simulations
+SEED = 9000
+
+class TestT1(unittest.TestCase):
+    """
+    Test measurement of T1
+    """
+
+    def test_t1(self):
+        """
+        Test T1 experiment using a simulator.
+        Currently only verifies that there is no exception,
+        but does not verify accuracy of the estimate.
```

Comment:
Note this comment. Moreover, the test is incorrect - the simulation does not make longer delays noisier.

Reply:
It might be good to test the circuit generation and fitter seperately.
Test that when you generate circuits you get the right number, and that the delay parameter of the gate and metadata in each ones is correct.

For the fitter test you can manually create some count data and metadata using a T1 error model and check you fit close to the correct T1 value.

Reply:
I began to implement this, but the circuits testing resulted in pretty much writing again the circuits function. Moreover, the tests were lacking verification that everything plays together and nothing breaks in the middle. I'd also like to be able to easily apply different noise parameters and multiple qubits. Therefore I prefer a solution of a simple dedicated backend. I'm going to push it now so you can see how it looks like. Indeed, thanks to the end-to-end testing I spotted a bug that the number of shots did not pass (I see that the bug is fixed in another PR).

Reply:
I guess that users can expect from Aer to support thermal noise on the delay gate, which takes into account the delay time. Do you want that I open an Aer issue about it?

Reply:
It appears to me that the following has been decided:
1. Aer will be added to the development requirements for testing (and hence to the CI).
2. In addition to the full-stack test, there will be a separate test for the analysis class.

This is good, but for characterization tests like T1 and T2 Aer backend is not helpful, because it does not support noise on delay gates as a function of the delay time. I think it is a natural feature to expect from Aer, and the solution should be to implement it in Aer (and remove the dedicated backend from `test_t1`). @chriseclectic what do you think?

Reply:
@mtreinish Note that we'd like to add Aer to the development requirements